src/GitVersionCore/StringFormatWith.cs

```diff
@@ -8,7 +8,7 @@ namespace GitVersion
 
     static class StringFormatWithExtension
     {
-        private static readonly Regex TokensRegex = new Regex(@"{\w+}", RegexOptions.Compiled);
+        private static readonly Regex TokensRegex = new Regex(@"{\$??\w+(\?\?\w+)??}", RegexOptions.Compiled);
```

Comment:
Since this is based on a regex, I would love to see some negative tests as well that challenges the regex engine a bit, just to make sure we don't have a too greedy expression or something that causes performance problems.

Reply:
Addressed with some additional testcases